pkg/resources/aws/aws.go

```diff
@@ -1324,6 +1324,10 @@ func FindNatGateways(cloud fi.Cloud, routeTables map[string]*resources.Resource,
 		}
 		response, err := c.EC2().DescribeNatGateways(request)
 		if err != nil {
+			if awsup.AWSErrorCode(err) == "NatGatewayNotFound" {
```

Comment:
If we pass in multiple nat gateway IDs, this will assume that all of them are already deleted, correct? Perhaps we should break this out into separate DescribeNatGateway calls per natGatewayId? I worry that we could leave orphaned NAT Gateways lying around which cost ~$30 USD/month. The old behavior made this situation more visible to the user since the kops command would fail, but now a user may not notice this situation.

Reply:
I'll run a test.

Reply:
It never occurred to me that the API could behave that badly. Rewritten.